lib/msf/ui/console/command_dispatcher/core.rb

```diff
@@ -1794,6 +1836,84 @@ def cmd_sessions(*args)
     true
   end
 
+  def get_matching_sessions(search_term)
+    matching_sessions = {}
+    terms = search_term.split
+    terms.each do |term|
+      matches = filter_sessions_by_search(term)
+      matches.each do |session_id, session|
+        matching_sessions.store(session_id, session)
+      end
+    end
+    matching_sessions
+  end
+
+  def filter_sessions_by_search(search_term)
+    matching_sessions = {}
+    
+    framework.sessions.each do |session_id, session|
+      case search_term.split(":")[0]
```

Comment:
Let's make sure to check if the search term is valid and feed that back to the user